tools/go.mod

```diff
@@ -24,17 +24,17 @@ require (
 	github.com/pingcap/kvproto v0.0.0-20240222024302-881fcbf5bc41
 	github.com/pingcap/log v1.1.1-0.20221110025148-ca232912c9f3
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.18.0
-	github.com/spf13/cobra v1.0.0
+	github.com/prometheus/client_golang v1.19.0
+	github.com/spf13/cobra v1.8.0
```

Comment:
ci failed because msg not contains "fail" https://github.com/tikv/pd/blob/0419004b7b4f0bef4b2cd13da568362805598f55/tools/pd-ctl/tests/keyspace/keyspace_test.go#L103

 https://github.com/tikv/pd/actions/runs/8532798384/job/23374503402?pr=8017

seems like after updating this library, had archive right redirect to error (maybe fix in v1.8.0?)

Maybe you need to add `root.SetErr(buf)` in https://github.com/tikv/pd/blob/ddf810be7ded85eaea9eef20125911bed058afb7/tools/pd-ctl/tests/helper.go#L29

cc @rleungx

Reply:
fixed in https://github.com/tikv/pd/pull/8021